frontend/dockerfile/docs/reference.md

```diff
@@ -818,6 +828,233 @@ The cache for `RUN` instructions can be invalidated by [`ADD`](#add) and [`COPY`
 
   If your system doesn't have support for `dirperm1`, the issue describes a workaround.
 
+## RUN --mount
+
+To use the `--mount` flag in `RUN` command, set Dockerfile version to at least
+[`1.2`](#syntax):
+
+```dockerfile
+# syntax=docker/dockerfile:1.4
+```
+
+`RUN --mount` allows you to create mounts that process running as part of the
+build can access. This can be used to bind files from other part of the build
+without copying, accessing build secrets or ssh-agent sockets, or creating cache
+locations to speed up your build.
+
+### RUN --mount=type=bind
+
+> **Note**
+>
+> `type=bind` is the default mount type being used if not provided
```

Comment:
Perhaps we should add a table under `RUN --mount` with the available types, and a "default", something like:

```markdown
Syntax:

    --mount=[type=<TYPE>][,option=<value>[,option=<value>]...]

### Mount types

| Type                                     | Description                                                                                               |
|------------------------------------------|-----------------------------------------------------------------------------------------------------------|
| [`bind`](#run---mounttypebind) (default) | Bind-mount context directories (read-only).                                                               |
| [`cache`](#run---mounttypecache)         | Mount a temporary directory for to cache directories for compilers and package managers..                 |
| [`secret`](#run---mounttypesecret)       | Allow the build container to access SSH keys via SSH agents, with support for passphrases.                |
| [`ssh`](#run---mounttypessh)             | Allow the build container to access secure files such as private keys without baking them into the image. |
```